src/thor/multimodal.cc

```diff
@@ -454,6 +445,18 @@ bool MultiModalPathAlgorithm::ExpandForward(GraphReader& graphreader,
           continue;
         }
       }
+
+      // Regular edge - use the appropriate costing and check if access is allowed
+      // and the walking distance didn't exceed (can't check in Allowed())
+      if (!mode_costing[static_cast<uint32_t>(mode_)]->Allowed(directededge, is_dest, pred, tile,
+                                                               edgeid, 0, 0, restriction_idx) ||
+          walking_distance > max_walking_dist_) {
+        continue;
+      }
+
+      Cost c = mode_costing[static_cast<uint32_t>(mode_)]->EdgeCost(directededge, tile);
+      c.cost *= mode_costing[static_cast<uint32_t>(mode_)]->GetModeFactor();
+      newcost += c;
```

Comment:
the rest of the changes in this block are just re-ordering: do all the checks where we might skip this edge candidate before actually getting to the cost